azurelinuxagent/daemon/scvmm.py

```diff
@@ -34,21 +35,35 @@ class ScvmmHandler(object):
     def __init__(self):
         self.osutil = get_osutil()
 
-    def detect_scvmm_env(self):
+    def isNotEmpty(self, list):
+        return list is not None
+
+    def detect_scvmm_env(self, dev_dir='/dev'):
         logger.info("Detecting Microsoft System Center VMM Environment")
-        self.osutil.mount_dvd(max_retry=1, chk_err=False)
+        found=False
+
+        # try to load the ATAPI driver, continue on failure
+        self.osutil.try_load_atapiix_mod()
+
+        # cycle through all available /dev/sr*|hd*|cdrom*|cd* looking for the scvmm configuration file
         mount_point = conf.get_dvd_mount_point()
-        found = os.path.isfile(os.path.join(mount_point, VMM_CONF_FILE_NAME))
-        if found:
-            self.start_scvmm_agent()
-        else:
-            self.osutil.umount_dvd(chk_err=False)
+        for devices in filter(self.isNotEmpty, [re.match(r'(sr[0-9]|hd[c-z]|cdrom[0-9]?|cd[0-9]+)', dev) for dev in os.listdir(dev_dir)]):
```

Comment:
@hglkrijger we only support python 3, though?

I find extra method not so clean, you can also do `lambda x: x is not None` in the filter.